src/lib.rs

```diff
@@ -314,6 +314,24 @@ impl<T> Sender<T> {
             _ => unreachable!(),
         }
     }
+
+    /// Consumes the Sender, returning a raw pointer.
+    ///
+    /// This call will leak memory unless the pointer is passed back to [Sender::from_raw].
+    pub fn into_raw(self) -> *mut () {
+        self.channel_ptr.as_ptr() as *mut ()
+    }
+
+    /// Consumes a raw pointer from [Sender::into_raw], recreating the Sender
+    ///
+    /// SAFETY: This pointer must have come from [Sender::into_raw] and must not have been passed
+    /// to `from_raw` before.
```

Comment:
> ... and must not have been passed to `from_raw` before.

This formulation is not wrong, but I think it can be expressed better. I'm sort of thinking about code that for some reason passes via into_raw -> from_raw -> into_raw -> from_raw multiple times. They pass the same pointer to `from_raw` multiple times. In essence, what is forbidden is to have multiple `Sender` instances for the same backing channel exist at any point in time, right?

Reply:
Good point.  I re-wrote it to be more precise, although I'm a bit worried that it's less clear for simple cases.  What do you think?